### PR TITLE
bpo-43109: Use /usr/bin/ar on macOS for LTO builds

### PR DESCRIPTION
**Pull request by xrisk**
https://bugs.python.org/issue43109

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1392,6 +1392,11 @@ if test "$Py_LTO" = 'true' ; then
           LLVM_AR_FOUND=found
           AC_MSG_NOTICE([llvm-ar found via xcrun: ${LLVM_AR}])
         fi
+        # /usr/bin/ar on macOS works fine in place of llvm-ar
```
Comment (matrixise) [inline configure.ac]: Here, you will just override the LLVM_AR variable, then the previous value was computed. You have to use a `else`.

Other question, could you check that `/usr/bin/ar` exists...

@ned-deily what's your opinion?

**Review by matrixise — CHANGES_REQUESTED** [review]

**Comment by ned-deily** [comment]
Thanks for the PR, @xrisk, and the heads up, @matrixise.  I will review and test this PR soon.

**Review by matrixise — COMMENTED** [review]
Thank you for your modifications, now, we have to wait for a review of @ned-deily

**Comment by vstinner** [comment]
cc @serge-sans-paille

**Comment by vstinner** [comment]
PR #26341 was merged instead.

**Pull request closed without merging**